monitoring/dashboards/dashboard.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/tidwall/gjson"
+	"github.com/tidwall/sjson"
+	"github.com/wushilin/stream"
+)
+
+const (
+	datasourceName = "dm-cluster"
+	outputDir      = "./dashboards"
+	name           = "dm.json"
+)
+
+var (
+	dashboards = map[string]string{
+		"dm.json": "Test-Cluster-DM",
+	}
+)
+
+func readDashboard(dir string, name string) (string, error) {
+	file := filepath.Join(dir, name)
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("File %s do not exists", file))
+	}
+	return string(data), nil
+}
+
+func writeDashboard(dir string, name string, body string) error {
+	title, exist := dashboards[name]
+	if !exist {
+		return errors.New(fmt.Sprintf("%s dashboard is not found in operator", name))
+	}
+
+	writeFile(dir, name, filterDashboard(body, name, title))
+	return nil
+}
+
+func writeFile(baseDir string, fileName string, body string) {
+	if body == "" {
+		return
+	}
+
+	fn := filepath.Join(baseDir, fileName)
+	f, err := os.Create(fn)
+	checkErr(err, "create file failed, f="+fn)
+	defer f.Close()
+
+	if _, err := f.WriteString(body); err != nil {
+		checkErr(err, "write file failed, f="+fn)
+	}
+}
+
+func checkErr(err error, msg string) {
+	if err != nil {
+		panic(errors.Wrap(err, msg))
+	}
+}
+
+func filterDashboard(body string, dashboard string, title string) string {
+	newStr := ""
+	stream.Of(body).Filter(func(str string) bool {
+		return str != ""
+	}).Map(func(str string) string {
+		// replace grafana item
+		r := gjson.Get(str, "__requires.0.type")
+		if r.Exists() && r.Str == "grafana" {
+			newStr, err := sjson.Set(str, "__requires.0.version", "")
```

Comment:
I think
```
			str, err = sjson.Set(str, "__requires.0.version", "")
```
is enough for this case, no need to use a stream package

Reply:
fixed